### ISSUE TITLE
kio-gdrive doesn't work, and a question about the upcoming plasma

### ISSUE THREAD

**Issue by pizzadude** [ZaWertun/fedora-copr-kde5#11]
Hi, first off, thanks for this copr repo.

I noticed if I use the google drive feature of Dolphin, kio-gdrive gives me this error:
![kio-gdrive-error](https://user-images.githubusercontent.com/1454420/66256764-347c9500-e75f-11e9-8d1a-2fcdbe26877d.png)
In the terminal dolphin shows this:
kf5.kio.core: couldn't create slave: "klauncher said: Error loading '/usr/lib64/qt5/plugins/kf5/kio/gdrive.so'."

Maybe add it to the copr repo and see if that fixes it?

P.S. Given the fact that you build kwin-lowlatency, and that will probably take a while to port to the new plasma coming out soon (5.17), will that affect your decision on updating this copr to 5.17, if it means you might have to drop kwin-lowlatency? https://github.com/tildearrow/kwin-lowlatency/issues/47 (I don't personally use kwin-lowlatency, but I am looking forward to plasma 5.17)

**Comment by ZaWertun** [comment]
kio-gdrive is built against old KF5 libs from official repo.
I'll build it in my copr - it will work then.

**Comment by pizzadude** [comment]
thanks :)

**Comment by ZaWertun** [comment]
Copr build: https://copr.fedorainfracloud.org/coprs/zawertun/kde/build/1047161/

**Comment by g6avk** [comment]
I think the kio-gdrive problem could be because google blocked it, much like the issues with Kmail and new accounts.

https://phabricator.kde.org/tag/kio_gdrive/

**Comment by ZaWertun** [comment]
> I think the kio-gdrive problem could be because google blocked it, much like the issues with Kmail and new accounts.
> 
> https://phabricator.kde.org/tag/kio_gdrive/

In current state kio-gdrive from F30 repo fails to load:
```
> kioclient5 ls 'gdrive:/'
kf5.kio.core: couldn't create slave: "klauncher said: Error loading «/usr/lib64/qt5/plugins/kf5/kio/gdrive.so»."
```

**Comment by ZaWertun** [comment]
Damn, also Qt (to 5.12.5) update in F31 breaks some KF5 libs, I'll rebuild them...

**Comment by g6avk** [comment]
Yep I should pay more attention,
https://bugs.kde.org/show_bug.cgi?id=411082

Sorry :-/

**Comment by pizzadude** [comment]
Weird, I haven't received that qt5 update in dnf yet and I use fedora 31. Maybe every mirror that dnf is querying is out of date or something...

**Comment by ZaWertun** [comment]
> thanks :)

Reinstall kio-gdrive from my repo:
`sudo dnf reinstall --repo zawertun-kde kio-gdrive`

At least it loads now. But I get some other error later... 🤔

**Comment by pizzadude** [comment]
it doesn't appear when doing dnf update --refresh and doing that command shows this:

sudo dnf reinstall --refresh --repo zawertun-kde kio-gdrive
Error: Unknown repo: 'zawertun-kde'

**Comment by ZaWertun** [comment]
Sorry, I'll think you're using F30, just wait till I rebuild KF5 libs for F31.

**Comment by pizzadude** [comment]
ok

**Comment by ZaWertun** [comment]
> it doesn't appear when doing dnf update --refresh and doing that command shows this:
> 
> sudo dnf reinstall --refresh --repo zawertun-kde kio-gdrive
> Error: Unknown repo: 'zawertun-kde'
> 
> maybe the mirrors aren't up to date yet

Also you have some other name for my repo, check it with `sudo dnf repolist`.

**Comment by pizzadude** [comment]
yeah the name is "copr:copr.fedorainfracloud.org:zawertun:kde"

**Comment by g6avk** [comment]
FWIW, I manually downloaded it from your repo.  
Gdrive works fine now from Dolphin and the CLI - F30
Thanks..

**Comment by ZaWertun** [comment]
Please try update again, kio-gdrive now built for F31 too.
(Also I didn't forget to refresh repo data now :)

**Comment by ZaWertun** [comment]
https://copr.fedorainfracloud.org/coprs/zawertun/kde/build/1048133/

**Comment by pizzadude** [comment]
kio-gdrive works now, but I also get this error upon upgrading:

sudo dnf update --best
[sudo] password for pizzadude:               
Last metadata expiration check: 0:03:05 ago on Sat 05 Oct 2019 05:30:19 PM EDT.
Error: 
 Problem 1: cannot install the best update candidate for package kf5-akonadi-server-19.04.3-3.fc31.x86_64
  - nothing provides libQt5Core.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides libQt5Sql.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides qt5-qtbase(x86-64) = 5.12.4 needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
 Problem 2: package kf5-akonadi-server-mysql-19.08.1-1.fc31.x86_64 requires kf5-akonadi-server(x86-64) = 19.08.1-1.fc31, but none of the providers can be installed
  - cannot install the best update candidate for package kf5-akonadi-server-mysql-19.04.3-3.fc31.x86_64
  - nothing provides libQt5Core.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides libQt5Sql.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides qt5-qtbase(x86-64) = 5.12.4 needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
 Problem 3: package kdepim-runtime-libs-1:19.08.1-1.fc31.x86_64 requires kf5-akonadi-server(x86-64) >= 19.08.1, but none of the providers can be installed
  - cannot install the best update candidate for package kdepim-runtime-libs-1:19.04.3-2.fc31.x86_64
  - nothing provides libQt5Core.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides libQt5Sql.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides qt5-qtbase(x86-64) = 5.12.4 needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
 Problem 4: package kdepim-runtime-1:19.08.1-1.fc31.x86_64 requires kdepim-runtime-libs(x86-64) = 1:19.08.1-1.fc31, but none of the providers can be installed
  - package kdepim-runtime-libs-1:19.08.1-1.fc31.x86_64 requires kf5-akonadi-server(x86-64) >= 19.08.1, but none of the providers can be installed
  - cannot install the best update candidate for package kdepim-runtime-1:19.04.3-2.fc31.x86_64
  - nothing provides libQt5Core.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides libQt5Sql.so.5(Qt_5.12.4_PRIVATE_API)(64bit) needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64
  - nothing provides qt5-qtbase(x86-64) = 5.12.4 needed by kf5-akonadi-server-19.08.1-1.fc31.x86_64

**Comment by ZaWertun** [comment]
Sorry for that.
All required packages was already built with new Qt version.
But there will be some lag while copr refresh repository metadata (wait ~1-2 hours).

I try updating Fedora 31 from development copr repo - update ends without any error.
If you want to update now use this patch:
```diff
--- _copr:copr.fedorainfracloud.org:zawertun:kde.repo.old       2019-10-05 22:14:10.663493119 +0300
+++ _copr:copr.fedorainfracloud.org:zawertun:kde.repo   2019-10-05 22:05:27.681747357 +0300
@@ -1,6 +1,6 @@
 [copr:copr.fedorainfracloud.org:zawertun:kde]
 name=Copr repo for kde owned by zawertun
-baseurl=https://copr-be.cloud.fedoraproject.org/results/zawertun/kde/fedora-$releasever-$basearch/
+baseurl=https://copr-be.cloud.fedoraproject.org/results/zawertun/kde/fedora-$releasever-$basearch/devel/
 type=rpm-md
 skip_if_unavailable=True
 gpgcheck=1
```
Notice `devel/` at the URL end.
Just don't forget to switch back afterwards.

**Comment by pizzadude** [comment]
Thanks, the upgrade works after doing that.
Also, I fixed the issue of files not opening with kio-gdrive in Dolphin, it was because I had the old "kdelibs3" library installed, which had a /usr/bin/kioexec that was conflicting with Plasma 5's kioexec. I also had to manually remove /usr/bin/kioexec after removing kdelibs3 because it was not owned by any package. The only program requiring kdelibs3 was this old ISO mounting program that I don't use. (I found better ones)
